### DISCUSSION TITLE
Capture MSI Cancel Installation Event

### DISCUSSION THREAD

**Discussion by oriov** [wixtoolset/issues#7504] [Questions]
Hi, 
I'm using wix3(WixUI_Minimal UI) and trying to create a custom action upon a user cancel installation. All I'm trying to do is send an event to our system upon user cancellation. Currently, I have the following configuration:

```
<UI>
    <UIRef Id="WixUI_Minimal" />			
</UI>

<CustomAction Id="GenerateGuid" BinaryKey="MyCustomActionBinary" DllEntry="GenerateGuid" Execute="immediate" Return="check" />
		<CustomAction Id="InstallStartedEvent" BinaryKey="MyCustomActionBinary" DllEntry="MsiInstallStarted" Execute="immediate" Return="check" />
		<CustomAction Id="UninstallEvent" BinaryKey="MyCustomActionBinary" DllEntry="MsiUninstallEvent" Execute="immediate" Return="check" />


		<InstallExecuteSequence>
			<Custom Action="InstallStartedEvent" After="InstallInitialize">NOT Installed AND NOT UPGRADINGPRODUCTCODE</Custom>
			<Custom Action="GenerateGuid" Before="InstallFinalize">NOT Installed AND NOT UPGRADINGPRODUCTCODE</Custom>
			<Custom Action="UninstallEvent" After="InstallValidate">REMOVE="ALL"</Custom>
		</InstallExecuteSequence>
```

Can i just add a custom action to InstallExecuteSequence element with some condition that meets the cancel action? And if so, what is that condition?

**Comment by BMurri** [comment]
The only fully reliable way would be to only distribute your MSI package via an executable bootstrapper, ideally where you either embed or download the MSI package. Every other possible solution can be disabled, so it can't be counted on.
